logfire/_internal/main.py

```diff
@@ -532,7 +545,26 @@ def log(
         attributes = attributes or {}
         merged_attributes = {**stack_info, **attributes}
         if (msg := attributes.pop(ATTRIBUTES_MESSAGE_KEY, None)) is None:
-            msg = logfire_format(msg_template, merged_attributes, self._config.scrubber)
+            fstring_frame = None
+            if self._config.inspect_arguments:
+                fstring_frame = inspect.currentframe()
+                if fstring_frame.f_back.f_code.co_filename == Logfire.log.__code__.co_filename:  # type: ignore
+                    # fstring_frame.f_back should be the user's frame.
+                    # The user called logfire.info or a similar method rather than calling logfire.log directly.
+                    fstring_frame = fstring_frame.f_back  # type: ignore
```

Comment:
Can't we use some utility function we have already?

Reply:
Here 'user' includes us. We need the place where the actual logfire method was called, even if it's inside an integration.